psi4/driver/p4util/python_helpers.py

```diff
@@ -291,6 +299,7 @@ def _core_wavefunction_to_file(wfn, filename=None):
     }  # yapf: disable
 
     if filename is not None:
+        if not filename.endswith('.npy'): filename += '.npy'
```

Comment:
should probably do a Pathlib pass through the codebase sometime (after 1.3) to abstract away remaining ties to str filesnames/paths.